web/packages/extension/src/common.ts

```diff
@@ -32,14 +32,62 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class NumberInputOption implements OptionElement<number | null> {
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {
+        this.numberInput.addEventListener("input", () => {
+            this.numberInput.reportValidity();
+        });
+    }
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value(): number | null {
+        const ni = this.numberInput;
+        const num = ni.valueAsNumber;
+        if (Number.isNaN(num)) {
+            return null;
+        }
+        if (ni.min) {
+            const min = Number.parseFloat(ni.min);
+            if (min > num) {
+                return min;
+            }
+        }
+        if (ni.max) {
+            const max = Number.parseFloat(ni.max);
+            if (num > max) {
+                return max;
+            }
+        }
+        return num;
+    }
+
+    set value(value: number | null) {
+        if (value === null) {
+            this.numberInput.value = "";
+        } else {
+            this.numberInput.value = value.toString();
+        }
+    }
+}
+
 /**
  * Number input option for 'maxExecutionDuration'.
  */
 class MaxExecutionDurationOption implements OptionElement<Duration> {
     constructor(
         private readonly numberInput: HTMLInputElement,
         readonly label: HTMLLabelElement
-    ) {}
+    ) {
```

Comment:
For code reuse, I suggest making `MaxExecutionDurationOption` a subclass of `NumberInputElement`:
```ts
class MaxExecutionDurationOption extends NumberInputElement implements OptionElement<Duration> { ... }
```

Also, how about renaming it to just `DurationOption`? At this point it doesn't really tied specifically to max execution time.

Reply:
In regards to making it a subclass, the type `OptionElement<Duration>` cannot currently be a subtype of `OptionElement<number | null>` as long as `ObsoleteDuration` is still one possible case of `Duration`. If we take a chance and guess that `ObsoleteDuration` can never occur in the extension 'option' UI, then I would guess that we could use `NumberDuration` instead and then use `NumberOption` as the superclass.

In regards to renaming it for general duration usage, I believe that instead of using  `MaxExecutionDurationOption`, that the class `NumberOption` would be possible to use there instead for new duration-based settings that do not use `Duration` but only `number`/`SecsDuration`. I believe it would be a good idea if any new duration-based setting only use `number` for its type, not `Duration`. I am also using an element ID at https://github.com/ruffle-rs/ruffle/blob/d92e4c3487f07f3f0d0953e8af85440171481165/web/packages/extension/src/common.ts#L140 , but that should be possible to refactor, or alternatively just add another ID-check.

**Question:** Should we take a chance and assume that `ObsoleteDuration` can never occur in the extension 'option' UI? I do not at all have a good feel for whether it can be possible or not for that to occur. If that chance is taken, then I think `MaxExecutionDurationOption` can be removed.

**I have renamed the Rust deserialization function `deserialize_max_execution_duration` to `deserialize_duration` to make it clearer that it can be used for any duration deserialization.** I have also added a comment discouraging usage of `Duration` in favor of `number`/`SecsDuration`.

Reply:
Great analysis. With that being said, I feel confidentally that `ObsoleteDuration` can never occour when configured from the extension UI, because `get value` always returns a number. So I think your conclusion is right and `MaxExecutionDurationOption` can be removed completely.

By the way, I like the renaming of `deserialize_max_execution_duration` to `deserialize_duration`!

Reply:
I have removed `MaxExecutionDurationOption`, and `NumberOption` is used instead.